### ISSUE TITLE
[FEAT] Large remote sensing image prediction (Tiff format, TTA, very large)

### ISSUE THREAD

**Issue by gy93-rs** [lightly-ai/lightly-train#385]
This is a truly convenient tool I've never encountered before. I only need to specify the dataset path to fine-tune a reasonably good semantic segmentation model on the DINOV3(vitl-sat493M) backbone.

As a researcher working in remote sensing imagery, I'd like to know how to better reason about images in my applications. For example, how to perform multi-scale reasoning (TTA), process TIFF images (specific to satellite imagery), and easily handle large images (over 10,000 x 10,000 pixels)?
I'm currently trying to modify the code in DINOv3EoMTSemanticSegmentation to meet all or some of these requirements.

I'd also like to know if there are plans to support alternative semantic segmentation decoders, particularly the mask2former. I understand that DINOv3 + mask2former might achieve even better accuracy.

**Comment by liopeer** [comment]
Hi @gy93-rs and great that you like LightlyTrain!

If you'd like to use the `DINOv3EoMTSemanticSegmentation` class for some downstream task (like doing predictions on multiple scales, using your own dataloaders with it (for TIFF images) or even integrate it into your internal training pipeline for multi-scale finetuning you can use the `lightly_train.load_model_from_checkpoint` function. This will actually return an instance of `DINOv3EoMTSemanticSegmentation` in your case that you can pluck into your code. The usage of the function is documented here: https://docs.lightly.ai/train/stable/semantic_segmentation.html#load-the-trained-model-from-checkpoint-and-predict

Be advised that there is currently an issue with loading SAT493m-based models. The issue is mentioned in #289 and I think it should be solvable by appending the correct hash to your exported checkpoint: `exported/exported_best.pt` --> `exported/exported_best-eadcf0ff.pt`. The issue stems from the following very heuristic logic in the DINOv3 source code:

https://github.com/lightly-ai/lightly-train/blob/5b1db20cbf54814dd140aa5d0612dfbd330dc5cf/src/lightly_train/_models/dinov3/dinov3_src/hub/backbones.py#L335-L362

> I'd also like to know if there are plans to support alternative semantic segmentation decoders, particularly the mask2former. 

We have no immediate plans to integrate Mask2Former, since we think EoMT has the really nice property of being super fast and almost as good in terms of performance (at least for the larger ViT variants), but I think we would definitely welcome a contribution for it! If you'd like to work on let me know and I will be happy to guide you through the process.

**Comment by yutong-xiang-97** [comment]
@gy93-rs and we also have TIFF support planned in the next few weeks :)

**Comment by gy93-rs** [comment]
@liopeer @yutong-xiang-97 Thank you for your replies.
Regarding the model weight loading issue, I addressed it by referring to the `load_train_state_dict` function you defined in #289  .

I've implemented a simple method for inference on large JPG images. For example, with a 10000*10000 pixel image, I used a 512*512 window to slide and crop across the large image (overlap=0.3) and batch inference (batchsize=16), which took only 46 seconds. By the way, my GPU is an NVIDIA 5090. I will try using TIFF format images once I obtain TIFF data. It's expected that the process will be similar. @yutong-xiang-97 

I haven't tried TTA yet. I trained the model using the default hyperparameters, only changing the batch size. I'm unsure how data augmentation was set during training. If possible, could you recommend some inference scales worth trying, such as [0.5, 0.75, 1, 1.25, 1.5]? Also, is folding augmentation necessary? If I want to utilize TTA, do I need to retrain for multi-scale augmentation?

Regarding the integration of the mask2former model, I plan to try it after I finish my recent work; I'm unsure if it will be successful.

**Comment by liopeer** [comment]
> I will try using TIFF format images once I obtain TIFF data. It's expected that the process will be similar.

Yes, the process will be exactly the same, but it will only be supported in a couple of weeks. :) The reason we want to support TIFF is that we see a lot of interest from the remote sensing community and the medical imaging community where TIFF/GeoTIFF is quite widely used.

> I'm unsure how data augmentation was set during training.

You can find the configuration here:
https://github.com/lightly-ai/lightly-train/blob/1f7f20c087d7291d7cec9deaf7da8574921f7bbc/src/lightly_train/_task_models/dinov3_eomt_semantic_segmentation/transforms.py#L68-L90

and in the docs: https://docs.lightly.ai/train/stable/semantic_segmentation.html#default-image-transform-arguments

This can be overriden by passing a dictionary to the `transform_args` parameter of the `train_semantic_segmentation` method (also described in the docs).

If you would like to do TTA, you would need to do this manually inside the `DINOv3EoMTSemanticSegmentation`. We currently don't support anything beyond resizing and normalizing there out-of-the-box. Maybe this is something that we might have to do more flexibly in the future.

> f possible, could you recommend some inference scales worth trying, such as [0.5, 0.75, 1, 1.25, 1.5]? Also, is folding augmentation necessary? If I want to utilize TTA, do I need to retrain for multi-scale augmentation?

Unfortunately I have very little experience with this, so I can't really give you a good answer I think. However, I am pretty sure that DINOv3 is robust to a lot of distortions, but your segmentation head might not be since it's only trained on a relatively small corpus of data.
